crates/wasm-encoder/src/core/code.rs

```diff
@@ -575,11 +587,24 @@ pub enum Instruction<'a> {
         array_type_index: u32,
         array_elem_index: u32,
     },
-
     RefTestNonNull(HeapType),
     RefTestNullable(HeapType),
     RefCastNonNull(HeapType),
     RefCastNullable(HeapType),
+    BrOnCast {
+        relative_depth: u32,
+        from_type_nullable: bool,
+        from_heap_type: HeapType,
+        to_type_nullable: bool,
+        to_heap_type: HeapType,
```

Comment:
This would mean enforcing the index implementation limit during decoding rather than validation, but I think that's fine and I think we do it elsewhere already, so not a big deal.